pkg/tap/switch.go

```diff
@@ -196,6 +196,9 @@ loop:
 		if err != nil {
 			return errors.Wrap(err, "cannot read size from socket")
 		}
+		if n != len(sizeBuf) {
+			return fmt.Errorf("unexpected size %d != %d", n, len(sizeBuf))
+		}
```

Comment:
What potential issue is this fixing?

https://golang.org/pkg/io/#ReadFull says:

> On return, n == len(buf) if and only if err == nil.

So this looks like a false positive in the linter you used?

Reply:
yes I removed this part